log-collector-script/linux/eks-log-collector.sh

```diff
@@ -189,9 +188,22 @@ create_directories() {
   done
 }
 
-get_instance_metadata() {
-  readonly INSTANCE_ID=$(curl --max-time 3 -H "X-aws-ec2-metadata-token: $TOKEN" --silent http://169.254.169.254/latest/meta-data/instance-id 2>/dev/null)
-  echo "${INSTANCE_ID}" > "${COLLECT_DIR}"/system/instance-id.txt
+get_instance_id() {
+
+  INSTANCE_ID_FILE="/var/lib/cloud/data/instance-id"
+
+  readonly INSTANCE_ID=$(curl --max-time 10 --retry 5 http://169.254.169.254/latest/meta-data/instance-id)
```

Comment:
In your PR summary, you state that the script will "fall back to query EC2 instance metadata in case a EC2 type doesn't support Instance Id from a File.." but that is not what is happening here. You are always querying the instance metadata from IMDS here and falling back to reading the INSTANCE_ID_FILE when the IMDS call fails.

Reply:
My bad. You are correct. I don't want to break existing workflow. Hence, it queries for EC2 metadata and then falls back to file.

Reply:
I actually think it would be best if you checked for the instance-id file first and then fell back on connecting to IMDS if it does not exist :)

There's nothing about the existing workflow that you would break.